### DISCUSSION TITLE
[Papercut Idea] `assertDispatchedOnce` on Job/Event/Mail fakes.

### DISCUSSION THREAD

**Discussion by paulandroshchuk** [laravel/framework#38734] [Ideas]
TLDR; just want to get a feedback whether it might be a good idea to add a new assertion method.

---

Laravel has a great API to test queued stuff like jobs/events/etc. Everything you need to do is to "fake" the needed facade and assert the job dispatched using the `assertDisptched()` helper.

```php
Bus::fake();

// do some stuff

Bus::assertDispatched(ProcessPodcast::class);
```

With this method, you can even assert that the right model was passed to the job:
```php
// Assert that the job's been dispatched with the right model.
Bus::assertDispatched(ProcessPodcast::class, fn ($job) => $job->podcast->is($podcast));
```
and assert that the job was dispatched the needed number of times:
```php
// Assert it's been dispatched only once.
Bus::assertDispatched(ProcessPodcast::class, $times = 1);
```

And what I'm used to do is to test both:
* Right model passed to the job.
* Job was dispatched only once.

```php
$podcast = Podcast::factory()->create();

$this->post('podcasts/'.$podcast->getKey().'/process');

Bus::assertDispatched(ProcessPodcast::class, fn ($job) => $job->podcast->is($podcast));
Bus::assertDispatched(ProcessPodcast::class, $times = 1);
```

The "issue" here is that it takes two lines to test it, so I'm wondering, what you guys think about a new `assertDispatchedOnce` method? I can easily do a PR for this, just wanted to get some feedback first.

```php
Bus::assertDispatchedOnce(ProcessPodcast::class, fn ($job) => $job->podcast->is($podcast))
```
